src/tiled/mapview.cpp

```diff
@@ -214,8 +214,8 @@ void MapView::wheelEvent(QWheelEvent *event)
     auto *hBar = static_cast<FlexibleScrollBar*>(horizontalScrollBar());
     auto *vBar = static_cast<FlexibleScrollBar*>(verticalScrollBar());
 
-    if (event->modifiers() & Qt::ControlModifier
-        && event->orientation() == Qt::Vertical)
+    if ((event->modifiers() & Qt::ControlModifier && event->orientation() == Qt::Vertical)
+        ^ !Preferences::instance()->getZoomPref())
```

Comment:
This check can't be right because the `event->orientation() == Qt::Vertical` check is necessary regardless of whether zooming is the default.